fix(@example/blog): optimize hero images

cansinacarer opened the pull request:
## Changes

Apply the [image best practices described in the docs](https://docs.astro.build/en/guides/images/#images-in-content-collections) to hero images of `examples/blog`:

- update the blog schema to change heroImage type from string to image
- move the placeholder images from `/public` to `/src/assets`
- update the sample blog posts' frontmatter with the new heroImage paths
- update the templates to use the `<Image />` component for hero images
- update the about page hero image

## Testing

These changes were applied in GitHub Codespaces and tested by running `pnpm --filter @example/blog run test`.

## Docs

These changes make the hero images in the blog example comply with the practices recommended in the existing documentation. I don't think a documentation update is needed.

florian-lefebvre commented on examples/blog/src/pages/blog/index.astro:
```diff
@@ -94,7 +95,7 @@ const posts = (await getCollection('blog')).sort(
 							<li>
 								<a href={`/blog/${post.id}/`}>
-									<img width={720} height={360} src={post.data.heroImage} alt="" />
+									<Image width={720} height={360} src={post.data.heroImage} alt="" />
```
This causes the tests to fail because heroImage can be undefined so there should be a condition here. Can you check if the design looks still good if there's no hero image?

cansinacarer replied:
Sorry about that! That specific test is now passing but there are a few other tests failing and they seem to be unrelated to the changes I made:
- I think [this one](https://github.com/withastro/astro/actions/runs/15379569478/job/43268575683?pr=13881) is failing because it needs sharp in the dependencies. Would you like me to add it in this PR?
- I'm not sure what is wrong with [this test](https://github.com/withastro/astro/actions/runs/15379569478/job/43268575685?pr=13881), there is a warning about `smoke/docs`.

After adding a condition for the case of missing hero image, the posts list doesn't look good when a post has no hero image, as you guessed:

- The look of the list when the newest post doesn't have a hero image (makes the title of the newest post look like the page title for the rest):
  ![image](https://github.com/user-attachments/assets/52d0d50c-2a98-4519-a6af-f39ed687ce9e)
- The look when other random posts don't have a hero image:
  ![image](https://github.com/user-attachments/assets/23743278-8515-46ef-b641-78145fe34748)

How about a fallback image instead? In `BaseHead.astro`, the first placeholder was used for fallback, but this fallback only applies to the `<head>`:
```js
const { title, description, image = '/blog-placeholder-1.jpg' } = Astro.props;
```

So I tried doing the same in commit 1e45ab2. Now that same image is shown for the posts without a hero image:
![image](https://github.com/user-attachments/assets/ce5226ae-681b-4b97-9a71-a80ad1f91e71)


Looking at the snippet above, I noticed moving the images from `public` to `src/asssets/` requires this fallback path to be updated as well. I did that in d403683.

Let me know what you think!

florian-lefebvre reviewed (changes requested):
Awesome PR thanks! The tests are failing (see my comment)

florian-lefebvre reviewed (changes requested):
1. Instead of a fallback image, I think it makes sense to make it required
2. Feel free to add sharp to fix the smoke test

cansinacarer commented:
> 1. Instead of a fallback image, I think it makes sense to make it required
> 2. Feel free to add sharp to fix the smoke test

I don't think it is user-friendly to tell a user "You are not allowed to post on your own website unless you get yourself a hero image", but there you go: I removed the fallback, made the image required (3731853), and added sharp (8e79f4a).

Adding sharp to resolve prevent [this test](https://github.com/withastro/astro/actions/runs/15379569478/job/43268575683?pr=13881) from failing made [another test](https://github.com/withastro/astro/actions/runs/15393909499/job/43309751308?pr=13881) fail because this test doesn't see that sharp is required by a test.
